### DISCUSSION TITLE
Mentions Support. Useful?

### DISCUSSION THREAD

**Discussion by mmlngl** [shadcn-ui/ui#2755] [Ideas]
## Summary
Can we contribute a `Mentions` component?

## Background
My team use Shadcn-ui in our project.
We have a Comment thread feature. As expected, we wanted to have @ and # functionality.

Today, Shadcn has components to make forms (`TextArea`, `Input`). People have [made and published custom extensions](https://github.com/JaleelB/shadcn-tag-input). We could go down that route too, but I thought I'd check if the community would prefer a new PR first.

## Demo
Here's how it looks in our app.
Note: custom theme applied.
https://github.com/shadcn-ui/ui/assets/46807/304e67c4-3230-4429-84d7-5b323777be52

## Discussion
- Naturally you can configure everything: data source, rendered components, etc. We can get into all this when/if there's a PR.
- Inspired by [`react-mentions`](https://github.com/signavio/react-mentions)

let me knooooow!

**Comment by AlexanderMoroz** [comment]
<img width="443" alt="Снимок экрана 2024-02-15 в 15 34 08" src="https://github.com/shadcn-ui/ui/assets/11835202/77657501-2683-4c80-ab94-b692291add89">

That's a screenshot of WIP based on shadui dropdown-menu component and `contentEditable` div with textarea styles

**Reply by valstu** [reply]
Just googled if shadcn-ui has this feature and ended up here :) I think you should definitely publish this. 🤩

**Comment by ghost** [comment]
It would be great if you'd open-source it!

**Comment by techwraith** [comment]
I need this, so yes please!

**Comment by compradia** [comment]
please give me an idea how to build such a component, I am suffering with react mentions for custom styles.

**Comment by AlexanderMoroz** [comment]
guys, sorry, it's been a while since we asked this question.

We'll discuss this in the team internally and try to publish it as soon as possible. Not sure should it be another lib or PR to shadui....
@mmmoli @techwraith @alexisbouchez @valstu

**Comment by mmlngl** [comment]
Thanks for all the support!

My original intention was to see if the repo maintainers were interested in this. I don't think we've made enough of a splash to get their attention.

**Comment by nesrucodex** [comment]
Why the heck shadcn doesn't have layout components like radix?

**Comment by patrik-u** [comment]
Would love this feature as well.

**Comment by Qodestackr** [comment]
Why is this feature stalled :smile: :rocket: push whatever :disappointed:

**Comment by mmlngl** [comment]
Only cos shadcn/UI maintainers have more important problems.

We made this thread to raise awareness.

It hasn't been raised enough yet, so thank you for the post. Tell your
friends.

**Comment by rossanodr** [comment]
this would be awesome!

**Comment by jonkurtis** [comment]
would be very interested in you open sourcing this @mmmoli

**Comment by supnim** [comment]
@mmmoli come on bro share the toys

**Comment by SofianeDjellouli** [comment]
I need this feature

**Comment by NOBLES5E** [comment]
and the official documentation should stop lying : )
![image](https://github.com/user-attachments/assets/6fa01326-3cd6-43da-8ed6-bf42c9555aab)

**Comment by r1me75** [comment]
Hi everyone,
Here are two examples of projects that have implemented a Mention component using ShadCN:

- https://craft.mxkaske.dev/post/fancy-area
- https://www.diceui.com/docs/components/mention

Hope this helps!

**Comment by axel7083** [comment]
Still something that would be very nice to see in the official repository!

**Comment by sandeeeerr** [comment]
Still no component like this in the main branch?

**Comment by MAiKo26** [comment]
Is there a extensible mention system if I wanted to add more logic to it ?

**Comment by imrankhanpersonalai** [comment]
looking for shadcn variant. would be really cool addition.

**Comment by bruisedpayne** [comment]
come on, add this thing

**Comment by Prajwal-alphanimble** [comment]
you might wanna check out this discussion where I implemented a new resizable and dragable dashboard layout... this can be very helpfull to many developers in making analytic, sharable dashboard, the users can alter the dashboard based on their liking...

The disscussion link: https://github.com/shadcn-ui/ui/discussions/10466
